Build failing with LLVM 3.6.1

After the recent ARM fix, I ran into this one. Is it an LLVM version issue? We use LLVM 3.6.1 by default on ARM.

````
In file included from codegen.cpp:517:0:
cgutils.cpp: In function 'void jl_dump_shadow(char*, int, const char*, size_t, bool)':
cgutils.cpp:495:30: error: 'class llvm::TargetMachine' has no member named 'getDataLayout'
     clone->setDataLayout(TM->getDataLayout()->getStringRepresentation());
                              ^
codegen.cpp: At global scope:
codegen.cpp:353:18: warning: 'show_execution_point_func' defined but not used [-Wunused-variable]
 static Function *show_execution_point_func;
                  ^
````

--- a/src/cgutils.cpp
+++ b/src/cgutils.cpp
@@ -489,7 +489,7 @@ static void jl_dump_shadow(char *fname, int jit_model, const char *sysimg_data, 
     // now copy the module, since PM.run may modify it
     ValueToValueMapTy VMap;
     Module *clone = CloneModule(shadow_module, VMap);
-#ifdef USE_MCJIT
+#ifdef LLVM37
     // Reset the target triple to make sure it matches the new target machine
     clone->setTargetTriple(TM->getTargetTriple().str());
     clone->setDataLayout(TM->getDataLayout()->getStringRepresentation());

